cmd/thanos/query.go

```diff
@@ -441,13 +441,15 @@ func runQuery(
 		}
 		logger := log.With(logger, "component", component.Query.String())
 
-		opts, err := defaultGRPCServerOpts(logger, reg, tracer, srvCert, srvKey, srvClientCA)
+		met := grpc_prometheus.NewServerMetrics()
+		opts, err := defaultGRPCServerOpts(logger, reg, tracer, met, srvCert, srvKey, srvClientCA)
```

Comment:
We should append the options afterwards I think, instead of passing them in here.

Reply:
Could please clarify more. Should I extract instrumentation options out from this method and append them later?